### ISSUE TITLE
Improve navigation experience (Multiple ways)

### ISSUE THREAD

**Issue by adriannelee** [cds-snc/notification-planning#558]
## Description

A11y audit issue: Content on the site can only be found through navigation. Add a secondary way, such as a search function or site map. 

As a (user), I need to be able to navigate Notify in multiple ways so that I can locate content and features in a manner that best meets my needs. 

WHY are we building? Non existent a11y support. Meeting users needs for clear navigation of the app. Providing an alternative to the main navigation. 

WHAT are we building? Start with a site map that indexes all of our public pages. 

VALUE created by our solution. 

- **For the Notify team:** 
  - A shared understanding of Notify's information architecture. 
- **For our users:** 
  - A clearer Notify experience where they feel less lost. At least with the public pages

## Acceptance Criteria** (Definition of done) 

Given a Notify user is located in any part of Notify, when they want to switch to another task or progress in a task, then more than one way is available to locate a task within a set of tasks except where the task is the result of, or a step in, a process.

- [x] Site map can navigate to all public facing pages
- [x] Site map can navigate to "your account" or "your services" if logged in (maybe)
- [x] Site map can be accessed from all pages - in the footer probably
- [x] Site map items in groups are alphabetically ordered in FR and EN. 
- [x] Site map groups are manually ordered. (hard coded)
- [x] Should we included archived policy docs (Ask Melissa)
- [x] Added a item to Zenhub template to keep sitemap up to date
- [x] New firewall rules
- [x] Translations for page title and group labels. (Site map is Plan de site)
- [x] A cypress test that clicks every link on the sitemap to see if link and h1 match. 
- [x] Articles menus are updated with new sitemap link

## A11y
Issue group: Header
Visibility: 3
Evaluation: Does not support
Success Criteria: [2.4.5: Multiple Ways](https://www.w3.org/WAI/WCAG21/Understanding/multiple-ways.html)
Needs design: Interaction, Content

[2.4.5 definition in relation to apps](https://www.w3.org/TR/wcag2ict/#navigation-mechanisms-mult-loc)

## Additional considerations
- Location within site not always clear
   - When sending a test email, on the Review before sending page, Templates appears underlined in the site navigation, indicating that the user is in the Templates section of the site. However, once the email is sent, the underline moves to Dashboard on the Email sent page. This change is somewhat confusing.

* Privacy considerations

## QA Steps

- [ ] All links are working
- [ ] Making sure things are ordered as expected
- [ ] All footers link the site map
- [ ] Responsive layout
- [ ] A11y scanning - proper nested list semantics

**Comment by amazingphilippe** [comment]
## Search
- What do we to be searchable? Public docs. Core tasks. Template content?
- Searching templates would also benefit us as admins. Full-text search, not just the naming structure (folder and templates)
- Will there be filters. By service, by date, by documentation

## Dashboard
- Do we need a sitemap if we rejig the dashboard to be more findable. 
- Dashboard is the other way of navigation
- Probably an easier implementation than a search bar
- Detach dashboards from services. User centered dashboard. Customizable dashboard

## Sitemap ([what is a sitemap](https://exposureninja.com/training/guides/seo/what-is-a-sitemap/))
- Surfaces everything in one page. Tricky to do a site map on a dashboard.

**Comment by andrewleith** [comment]
Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Notify-Planning-614b3ad91bc2030015ed22f5/issues/cds-snc/notification-planning/558?planning-poker) with ZenHub @amazingphilippe

**Comment by yaelberger-commits** [comment]
Team will do a STT on Thursday

**Comment by jimleroyer** [comment]
Design is ready so we can pursue with implementation.

**Comment by yaelberger-commits** [comment]
This user story for A11y will not be completed for this set of OKRs, so we should disclose it in our Accessibility Statement.

**Comment by YedidaZalik** [comment]
@Andrew Leith shared the workflow below that he developed in a previous position:
I drew up something similar a while back, but it was out of scope (second attachment below).
Including for consideration when we work on this.

![image.png](https://images.zenhubusercontent.com/614b9f22b18cf806c628a310/5787dd84-7fad-4d3d-b6ad-13747842af65)

![Screenshot 2023-06-27 at 2.06.50 PM.png](https://images.zenhubusercontent.com/614b9f22b18cf806c628a310/b6f66a15-aeab-490c-9850-b20ce423dcbf)

**Comment by YedidaZalik** [comment]
Included in Ally stmt

**Comment by yaelberger-commits** [comment]
Can also use this as an opportunity to improve guidance navigation, either through the sitemap or additionally with an index

**Comment by amazingphilippe** [comment]
Designed an initial site map from our IA diagram. https://www.figma.com/file/r3GzRBjBDWvL2AwCWq1e1W/Navigating-the-site-%7C-Naviguer-le-site?type=design&node-id=8840-10&mode=design&t=sL6RrmwGJtduDARE-4

**Comment by YedidaZalik** [comment]
Phil and Yedida met yesterday
A couple ideas and can look at during dev design review
Could consider for story refinement

**Comment by amazingphilippe** [comment]
Moving to the backlog until we bring to dev design review

**Comment by andrewleith** [comment]
Most of the dev work is complete, now waiting on design to finalize content/design.

**Comment by YedidaZalik** [comment]
Met with Yael
System status page is also used by non-developers
- API documentation repeats in 2 sections so System status can repeat also
- Yael says put it under Help and guidance. I put it at the end of that section since it's different kind of help from the other topics
Guidance overview
- change overview to index here (Forms does not have a site map and just calls it Guidance)
Will add this to the Github card

**Comment by YedidaZalik** [comment]
Sent context to MSB and asked her to provide translation to Andrew (I'm off Thurs/Fri) for 

Your GC Notify
